### PR TITLE
letterpress: init at 2.1

### PR DESCRIPTION
**Pull request by dawidd6**
## Description of changes

As part of: https://github.com/NixOS/nixpkgs/issues/195280

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/le/letterpress/package.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  python3Packages,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "letterpress";
+  version = "2.1";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/le/letterpress/package.nix:26]:
```suggestion
    repo = "letterpress";
```

File: pkgs/by-name/le/letterpress/package.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    desktop-file-utils
+    gettext
+    gobject-introspection
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (Aleksanaa) [inline pkgs/by-name/le/letterpress/package.nix:60]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/le/letterpress/package.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  meta = with lib; {
+    description = "Create beautiful ASCII art";
+    longDescription = ''
+      Letterpress converts your images into a picture made up of ASCII characters.
+      You can save the output to a file, copy it, and even change its resolution!
+      High-res output can still be viewed comfortably by lowering the zoom factor.
+    '';
+    homepage = "https://apps.gnome.org/Letterpress/";
+    license = licenses.gpl3Plus;
```
Comment (Aleksanaa) [inline pkgs/by-name/le/letterpress/package.nix:80]: Please add `mainProgram`.

**Comment by Aleksanaa** [comment]
My preview isn't shown in monospace font, but I'm not sure if it's a font configuration problem. Are you the same?

image

**Comment by dawidd6** [comment]
I'm not sure if we should include this project in nixpkgs.

Project's author is reluctant in supporting other packaging formats (other than flatpak):
https://gitlab.gnome.org/World/Letterpress/-/merge_requests/16#note_2120586

Also the thumb down on this PR from them is telling.

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/le/letterpress/package.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  dependencies = with python3Packages; [
+    pillow
+    pygobject3
+  ];
+
+  format = "other";
```
Comment (Aleksanaa) [inline pkgs/by-name/le/letterpress/package.nix:65]:
```suggestion
  pyproject = false; # built by meson
```

Small suggestion, as `format = "other"` seems deprecated.

**Review by marcusramberg — APPROVED** [review]
lgtm. I get monospaced output in i3.

**Pull request merged**